va/tlsalpn.go

```diff
@@ -189,14 +196,19 @@ func (va *ValidationAuthorityImpl) validateTLSALPN01(ctx context.Context, identi
 
 	leafCert := certs[0]
 
-	// Verify SNI - certificate returned must be issued only for the domain we are verifying.
-	if len(leafCert.DNSNames) != 1 || !strings.EqualFold(leafCert.DNSNames[0], identifier.Value) {
+	// The certificate returned must have a subjectAltName extension containing
+	// only the dNSName being validated and no other entries.
+	if len(leafCert.EmailAddresses) != 0 ||
+		len(leafCert.IPAddresses) != 0 ||
+		len(leafCert.URIs) != 0 ||
+		len(leafCert.DNSNames) != 1 ||
+		!strings.EqualFold(leafCert.DNSNames[0], identifier.Value) {
```

Comment:
Code looks good, but we don't have coverage for this conditional. Could you extract into a helper and add a pass and fail for case for the helper?

Reply:
Tests added